### ISSUE TITLE
[bitnami/kube-state-metrics] Errors in prometheus when kube-state-metrics is deployed with multiple replicas

### ISSUE THREAD

**Issue by jw-maynard** [bitnami/charts#5143]
**Which chart**:
kube-state-metrics - 1.1.6

**Describe the bug**
Currently this chart allows you to run multiple replicas but the kube-state-metrics project is not designed to run multiple replicas unless you enable sharding (see https://github.com/kubernetes/kube-state-metrics#horizontal-scaling-sharding). This causes issues in Prometheus because 2 instances of kube-state-metrics collect the metrics for the same pod.

```
Error executing query: found duplicate series for the match group {namespace="kube-system", pod="aws-node-4vqxr"} on the right hand-side of the operation:
[{
		__name__ = "kube_pod_info",
		container = "kube-state-metrics",
		created_by_kind = "DaemonSet",
		created_by_name = "aws-node",
		endpoint = "http",
		host_ip = "172.16.1.118",
		instance = "172.16.5.149:8080",
		job = "bitnami-kube-prometheus-kube-state-metrics",
		namespace = "kube-system",
		node = "ip-172-16-1-118.ec2.internal",
		pod = "aws-node-4vqxr",
		pod_ip = "172.16.1.118",
		priority_class = "system-node-critical",
		service = "bitnami-kube-prometheus-kube-state-metrics",
		uid = "44fba93b-f67c-42f3-a158-f53d9613beaa"
	},
	{
		__name__ = "kube_pod_info",
		container = "kube-state-metrics",
		created_by_kind = "DaemonSet",
		created_by_name = "aws-node",
		endpoint = "http",
		host_ip = "172.16.1.118",
		instance = "172.16.1.139:8080",
		job = "bitnami-kube-prometheus-kube-state-metrics",
		namespace = "kube-system",
		node = "ip-172-16-1-118.ec2.internal",
		pod = "aws-node-4vqxr",
		pod_ip = "172.16.1.118",
		priority_class = "system-node-critical",
		service = "bitnami-kube-prometheus-kube-state-metrics",
		uid = "44fba93b-f67c-42f3-a158-f53d9613beaa"
	}
];
many-to-many matching not allowed: matching labels must be unique on one side
```

**Additional context**
It may be enough to just add something to the documentation to "fix" this issue. The documentation for the kube-prometheus chart might need to be updated also as this issue causes problems querying metrics when that's deployed with multiple kube-state metrics replicas.

**Comment by javsalgar** [comment]
Hi,

Thank you very much for the input! I agree that, if we allow mulitple replicas, we should enable the shard parameter. Could you confirm that, by using the `extraArgs` value with the missing parameters, then the multiple replicas work without issues? Then I'll create a PR.

**Comment by jw-maynard** [comment]
Hey @javsalgar I worked around this in a different way. I'm using the kube-prometheus chart which uses this chart as a sub chart. We don't have a particularly large cluster (it's tiny actually) and so the horizontal benefits of sharding really are not useful to us. Since this chart currently doesn't support sharding I added a relabel config like so
```
        relabelings:
            - action: replace  # Required if there are multiple replicas
              sourceLabels:
                  - __meta_kubernetes_endpoints_name
              targetLabel: instance
```
This makes all of the labels exactly the same for each endpoint (in our case 2), and the query problem goes away.

I also did some research on the Prometheus user group and it appears that if all the labels of a series match entirely inside any given scrape job Prometheus will dedup the data automatically so we're not even pulling twice the data points for each kube-state-metrics time series.

Given all of this new information I think it would make sense to make it a feature request to add the possibility to run statefulsets for people who need horizontal scaling, but my problem is solved with this configuration. Maybe just adding the above to the documentation is enough to "resolve" this "bug" with using replicas? If you agree I can try to make a PR to update the README.md
